src/ui/components/statusrow/StatusRow.js

```diff
@@ -81,7 +87,8 @@ const STATUSES = [
 ]
 
 const StatusRow = (props) => {
-    const generateCells = (entity, entityType, hasHazards) => {
+    const generateCells = (entity, entityType, safetyData) => {
+        const hasHazards = processSafetyData(safetyData);
```

Comment:
You could just do an await of the fetchSafetyData() here

Reply:
Again, I would have to change it to an async function and it'll break the result, I already tried.